File: dataset/ourMethod/OPENAI_gpt-5-2025-08-07/Task155.py
from typing import List, Sequence

MAX_LEN = 1_000_000

def heap_sort(data: Sequence[int]) -> List[int]:
    if data is None:
        raise ValueError("Input must not be None.")
    if not isinstance(data, (list, tuple)):
        raise ValueError("Input must be a list or tuple of integers.")
    n = len(data)
    if n < 0 or n > MAX_LEN:
        raise ValueError("Invalid input length.")
    for i, x in enumerate(data):
        if not isinstance(x, int):
            raise ValueError(f"All elements must be integers. Invalid at index {i}.")

    arr: List[int] = list(data)

    def sift_down(start: int, heap_size: int) -> None:
        root = start
        while True:
            left = 2 * root + 1
            if left >= heap_size:
                break
            right = left + 1
            largest = root
            if arr[left] > arr[largest]:
                largest = left
            if right < heap_size and arr[right] > arr[largest]:
                largest = right
            if largest == root:
                break
            arr[root], arr[largest] = arr[largest], arr[root]
            root = largest

    # Build max heap
    for i in range(n // 2 - 1, -1, -1):
        sift_down(i, n)

    # Extract elements
    for end in range(n - 1, 0, -1):
        arr[0], arr[end] = arr[end], arr[0]
        sift_down(0, end)

    return arr


def _print_array(a: List[int]) -> None:
    print("[" + ", ".join(str(x) for x in a) + "]")


if __name__ == "__main__":
    try:
        tests = [
            [3, 1, 4, 1, 5, 9, 2, 6, 5],
            [],
            [42],
            [1, 2, 3, 4, 5],
            [0, -1, -3, 2, -1, 2, 0],
        ]
        for t in tests:
            _print_array(heap_sort(t))
    except ValueError as e:
        print(f"Error: {e}")

File: dataset/ZeroShot/OPENAI_gpt-5-2025-08-07/Task155.cpp
#include <bits/stdc++.h>
using namespace std;

class Task155 {
public:
    static vector<int> heap_sort(const vector<int>& arr) {
        vector<int> a = arr;
        const size_t n = a.size();

        auto sift_down = [&](size_t heap_size, size_t i) {
            while (true) {
                size_t largest = i;
                size_t left = i * 2 + 1;
                size_t right = left + 1;

                if (left < heap_size && a[left] > a[largest]) largest = left;
                if (right < heap_size && a[right] > a[largest]) largest = right;
                if (largest == i) break;
                std::swap(a[i], a[largest]);
                i = largest;
            }
        };

        if (n > 1) {
            for (size_t i = (n / 2); i-- > 0; ) {
                sift_down(n, i);
            }
            for (size_t end = n - 1; end > 0; --end) {
                std::swap(a[0], a[end]);
                sift_down(end, 0);
            }
        }
        return a;
    }
};

static void print_vec(const vector<int>& v) {
    cout << "[";
    for (size_t i = 0; i < v.size(); ++i) {
        if (i) cout << ", ";
        cout << v[i];
    }
    cout << "]\n";
}

int main() {
    vector<vector<int>> tests = {
        {},
        {1},
        {5, 3, 8, 4, 2, 7, 1, 10},
        {1, 2, 3, 4, 5},
        {-3, -1, -7, 0, 2, 2, -3}
    };

    for (const auto& t : tests) {
        auto sorted = Task155::heap_sort(t);
        print_vec(sorted);
    }
    return 0;
}